libyul/backends/evm/EVMDialect.cpp

```diff
@@ -383,6 +390,7 @@ std::vector<std::optional<BuiltinFunctionForEVM>> createBuiltins(langutil::EVMVe
 			1,
 			1,
 			SideEffects{},
+			ControlFlowSideEffects{},
```

Comment:
Like I mentioned in https://github.com/ethereum/solidity/pull/15512#discussion_r1809364632, I'd also extract this bit from `createEVMFunction()` into a helper: https://github.com/ethereum/solidity/blob/a80a4daaa8192d9cf5f7c164a7c591260efc3d68/libyul/backends/evm/EVMDialect.cpp#L71-L84

This would let us avoid setting them by hand and getting them wrong. They can already be derived from the info we have in `SemanticInformation` already, and that should be the single source of truth.

Then here you would do:
```suggestion
			ControlFlowSideEffects::fromInstruction(evmasm::Instruction::DATALOADN),
```

Reply:
Added the helper. One comment here. It won't be always an instruction which can be used to get proper `ControlFlowSideEffects`. If we want to use this helper with `setimmutable` for legacy dialect it's not clear which instruction should be pass to the helper.

Reply:
But the helper you added does not have the logic I linked above. Is it the final version?

> If we want to use this helper with `setimmutable` for legacy dialect it's not clear which instruction should be pass to the helper.

Yes, of course. The helper is meant only for those builtins that are based on an instruction. That's the case with `auxdataloadn()`. And will be the case with `eofcreate()` and `returncontract()`. Maybe with more of them.

Reply:
Sorry. Too tired yesterday night. Fixed now